src/outputs/form_output_prep.py

```diff
@@ -34,6 +35,13 @@ def form_output_prep(
         estimated_df["imp_marker"] == "R"
     )
 
+    # Deal with "No" in 604, also eliminating spenders
+    flag_no_rand_spenders(estimated_df, "warn")
+    no_rnd_spenders_filter = ~(
+        (estimated_df["604"] == "No") & (estimated_df["211"] > 0)
+    )
+    estimated_df = estimated_df.copy().loc[no_rnd_spenders_filter]
```

Comment:
I think it might make more sense to put this in after imputation. It's only in imputation that it's possible to get these contradictory records, since we eliminated them in staging. If we leave it here, we'll have a difference between estimation_qa outputs and final outputs.

Reply:
I think we've got 2  choices here: either we decide that if it's "check needed" and "604 = No", then we re-set the imputed values to zero (or better, don't impute them).
Alternatively, we should probably change "604" to "Yes" if it's TMI, like we do for MoR. I suggest we do this second thing for now so it will be easy to see what remains of the inconsistency